libraries/AP_ESC_Telem/AP_ESC_Telem.cpp

```diff
@@ -131,7 +131,7 @@ bool AP_ESC_Telem::are_motors_running(uint32_t servo_channel_mask, float min_rpm
         if (BIT_IS_SET(servo_channel_mask, i)) {
             const volatile AP_ESC_Telem_Backend::RpmData& rpmdata = _rpm_data[i];
             // we choose a relatively strict measure of health so that failsafe actions can rely on the results
-            if (now < rpmdata.last_update_us || now - rpmdata.last_update_us > ESC_RPM_CHECK_TIMEOUT_US) {
+            if ((rpmdata.last_update_us == 0) || ((now - rpmdata.last_update_us) > ESC_RPM_CHECK_TIMEOUT_US)) {
```

Comment:
Can't you just remove the check. The safe wrap pattern is 
```
((now - past_time) > timeout)
```
I believe this works correctly with warpping

Reply:
Not without leaving a hole for the first `ESC_RPM_CHECK_TIMEOUT_US` after wrap. We are early returning here, which means an unset time of 0 happens any time the data times out.